python/search_completer/hierarchical_search_completer.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+from .search_completer import SearchCompleter
+
+shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
+
+
+class HierarchicalSearchCompleter(SearchCompleter):
+    """
+    A standalone :class:`PySide.QtGui.QCompleter` class for matching SG entities to typed text.
+
+    :signal: ``node_activated(str, int, str, str, list)`` - Fires when someone activates a
+        node inside the search results. The parameters are ``type``, ``id``, ``name``,
+        ``label path`` and ``incremental_paths``. If the node activated is not an entity,
```

Comment:
the hierarchy API returns different `kinds` of things - wondering if there are usecases where this would be useful to have returned. it feels like the current signal is perfect for when an entity is clicked (the normal case) but if something else is clicked it's a bit of a black box.

How does this work practically? if there is a hierarchy node `Assets > Vehicle (sg asset type) > Ford`, can i search in the completer for `Vehicle`? 

If yes, i think it would be helpful to have a mode where the completer only returns entities, so a mode where we know that the `node_activated` signal always returns `type` and `id` that are never `None`. I think this is the normal use case.